OrbitLinuxTracing/UprobesReturnAddressManager.h

```diff
@@ -59,6 +59,34 @@ class UprobesReturnAddressManager {
     }
   }
 
+  void PatchCallchain(pid_t tid, uint64_t* callchain, uint64_t nr,
+                      unwindstack::Maps* maps) {
+    if (!tid_uprobes_stacks_.contains(tid)) {
+      return;
+    }
+
+    auto& tid_uprobes_stack = tid_uprobes_stacks_.at(tid);
+    CHECK(!tid_uprobes_stack.empty());
+
+    uint64_t i = 0;
+    // TODO(kuebler): What about tail-call optimization, where two uretprobes
+    //  hijacked an address at the same stack.
+    for (auto it = tid_uprobes_stack.rbegin(); it != tid_uprobes_stack.rend();
+         it++) {
+      const OpenUprobes& uprobes = *it;
+      for (; i < nr; i++) {
```

Comment:
Could you write this with a single loop where you go through the callchain and increment `it` when you encounter a uprobes IP? In other words, iterate over the callchain, not over `tid_uprobes_stack`? I think it would be easier to understand.

Reply:
I refactored it like that. Hope it is better now.